lib/Doctrine/Record.php

```diff
@@ -1593,6 +1593,7 @@ public function coreSetRelated($name, $value)
                 }
             } else {
                 $localFieldName = $this->_table->getFieldName($rel->getLocal());
+                $foreignFieldName = '';
```

Comment:
This will get set to something in the if condition below, but if that evaluates to false, wouldn't get set, and is then used on 1610 and 1617.